### PR TITLE
docs: add nginx unit section

### PR DESCRIPTION
**Pull request by Kumzy**
### Pull Request Checklist

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [ ] Pre-Commit Checks were ran and passed
- [ ] Tests were ran and passed

### Description
This add a new deployment folder in the documentation in the **usage** section as described in #2609.
Add a sub-section about using NGINX Unit to run litestar

### CODE REVIEW

File: docs/usage/deployment/nginx-unit.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
+NGINX Unit
+==========
+
+To run litestar using `nginx-unit <https://unit.nginx.org/>`_
+
+Install nginx-unit
+------------------
+
+1. To install nginx-unit, refers to the `official documentation <https://unit.nginx.org/installation/>`_
```
Comment (JacobCoffee) [inline docs/usage/deployment/nginx-unit.rst:9]:
```suggestion
To install ``nginx-unit``, refer to the `official documentation <https://unit.nginx.org/installation/>`_
```

File: docs/usage/deployment/nginx-unit.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+.. tab-set::
+
+   .. tab-item:: macOS (Brew)
```
Comment (JacobCoffee) [inline docs/usage/deployment/nginx-unit.rst:13]:
```suggestion
   .. tab-item:: macOS (`Brew <https://brew.sh/>`_)
```

File: docs/usage/deployment/nginx-unit.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    :caption: unit.json
+
+Listeners
++++++++++
+
+To accept requests, add a listener object in the config/listeners API section; the object’s name can be:
+
+- A unique IP socket: 127.0.0.1:80, [::1]:8080
+- A wildcard that matches any host IPs on the port: *:80
+- On Linux-based systems, abstract UNIX sockets can be used as well: unix:@abstract_socket.
```
Comment (JacobCoffee) [inline docs/usage/deployment/nginx-unit.rst:53]:
```suggestion
- A unique IP socket: ``127.0.0.1:80``, ``[::1]:8080``
- A wildcard that matches any host IPs on the port: ``*:80``
- On Linux-based systems, abstract UNIX sockets can be used as well: ``unix:@abstract_socket``.
```
sphinx will add a link to that `*`

File: docs/usage/deployment/nginx-unit.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
++------------------------+-------------------+----------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``processes``          | 1                 | Integer or object; integer sets a static number of app processes, and object options `max`, `spare`, and `idle_timeout` enable dynamic management. |
+|                        |                   |                                                                                                                                                    |
+|                        |                   | The default is 1.                                                                                                                                  |
++------------------------+-------------------+----------------------------------------------------------------------------------------------------------------------------------------------------+
+
+Configuration update
+--------------------
+
+To update the nginx-unit service already running, use ``PUT`` method to send the ``unit.json`` file on the `/config` endpoint
```
Comment (JacobCoffee) [inline docs/usage/deployment/nginx-unit.rst:94]:
```suggestion
To update the nginx-unit service already running, use ``PUT`` method to send the ``unit.json`` file on the ``/config`` endpoint
```

File: docs/usage/deployment/index.rst
Comment (JacobCoffee) [inline docs/usage/deployment/index.rst:1]: Can you move to docs/topics/deployment? Sorry 🙁

File: docs/usage/deployment/nginx-unit.rst
Comment (JacobCoffee) [inline docs/usage/deployment/nginx-unit.rst:1]: Same as above on moving to Topics section

**Review by JacobCoffee — CHANGES_REQUESTED** [review]

**Comment by JacobCoffee** [comment]
Everything looks good to me except the move from Usage to Topics! 

Thanks

File: docs/usage/deployment/index.rst
Comment (Kumzy) [reply docs/usage/deployment/index.rst:1]: Done

File: docs/usage/deployment/nginx-unit.rst
Comment (Kumzy) [reply docs/usage/deployment/nginx-unit.rst:1]: Done

File: docs/usage/deployment/index.rst
Comment (Kumzy) [reply docs/usage/deployment/index.rst:1]: Maybe the deployment section need a little introduction?

**Review by Kumzy — COMMENTED** [review]
Moved from usage to topic

**Review by Kumzy — COMMENTED** [review]
Forgot to review comments

**Review by JacobCoffee — APPROVED** [review]

**Comment by Alc-Alc** [comment]
@all-contributors add @Kumzy to docs

**Pull request merged**